### PR TITLE
[task1a1b] added imdb_movie_info_crwaler module

### PR DESCRIPTION
**Pull request by Lucas5Lucas**

### CODE REVIEW

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+/**
+ * @file
+ * This module crawls Imdb movie pages and displays the movie title, picture and movie plot. 
+ */
+
+/**
+ * Implements hook_menu().
+ */
+function imdb_movie_info_crawler_menu() {
+    $items['Imdb_Crawler'] = array(
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: use small letters

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      'title' => 'Imdb crawler',
+      'page callback' => 'drupal_get_form',
+      'page arguments' => array('imdb_movie_info_crawler_form'),
+      'access callback' => TRUE,
+      'type' => MENU_NORMAL_ITEM,
+    );
+    $items['Movie_Info'] = array(
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: same here

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+/**
+ * Implements hook_menu().
+ */
+function imdb_movie_info_crawler_menu() {
+    $items['Imdb_Crawler'] = array(
+      'title' => 'Imdb crawler',
+      'page callback' => 'drupal_get_form',
+      'page arguments' => array('imdb_movie_info_crawler_form'),
+      'access callback' => TRUE,
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: add access checks

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      'type' => MENU_NORMAL_ITEM,
+    );
+    $items['Movie_Info'] = array(
+      'title' => 'Movie Info',
+      'page callback' => 'movie_info_page_callback',
+      'access callback' => TRUE,
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: add access checks

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+/**
+ * Page callback for Imdb_Crawler path.
+ */
+function imdb_movie_info_crawler_form($form, &$form_state) {
+    $form['movie_url'] = array(
+      '#title' => t('Enter Imdb movie link'),
+      '#type' => 'textfield',
+      '#description' => 'Enter the imdb movie link here',
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: put t()

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+/**
+ * Implements hook_form_validate().
+ */
+function imdb_movie_info_crawler_form_validate($form, &$form_state) {
+    $submitted_url = $form_state['values']['movie_url'];
+    $string_pattern_url = "'http://www.imdb.com/title/tt'si";
+    $valid_url = preg_match($string_pattern_url, $submitted_url);
+    if (!$valid_url) {
+        drupal_set_message("Please enter a valid url!");
+        $form_state['rebuild'] = TRUE;
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: not needed

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    }
+}
+
+/**
+ * Implements hook_form_submit().
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: kein hook...

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+ */
+function imdb_movie_info_crawler_form_submit($form, &$form_state) {
+    $movie_infos = get_imdb_movie_infos($form_state['values']['movie_url']);
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: remanem get_imdb_movie_infos to imdb_movie_info_crwaler_fetch_infos()

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    $_SESSION['movie_infos'] = $movie_infos;
+    drupal_goto("Movie_Info");
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: Hinweise:

1) drupal_goto('url', array('query' => array('title' => $movie_infos['title'], ))
2) SeSSION
3) write result to db -> redirect with ID -> on new page load from db

4) Goto() -> $form_state['redirect'] ersetzten.

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+/**
+ * Page callback for Movie_Info path.
+ */
+function movie_info_page_callback() {
+    $movie_infos = array(
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: add checks for session var,

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+      'title' => $_SESSION['movie_infos']['title'],
+      'posterlink' => $_SESSION['movie_infos']['posterlink'],
+      'postersrc' => $_SESSION['movie_infos']['postersrc'],
+      'plot' => $_SESSION['movie_infos']['plot'],
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: eg.: 'plot' = !empty($_SESSION['movie_infos']['plot']) ? $_SESSION['movie_infos']['plot'] : '';

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+    );
+    $page['Movie_Info'] = [
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: lower case

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+/**
+ * Returns movie title, link to the posterlink, the source of the poster image and the movie plot.
+ *
+ * @params string $url
+ *   The movie url of an imdb movie page
+ */
+function get_imdb_movie_infos($url) {
+    @$homepage = file_get_contents($url);
+
+    //extracts movie title from page
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: coding convention
// Extracts movie tile from page.

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+    preg_match($string_pattern_poster_html, $homepage, $regextextposter);
+    $string_pattern_poster_link = "'href=\"(.*?)\"'si";
+    preg_match($string_pattern_poster_link, $regextextposter[0], $posterlink);
+    $string_pattern_poster_src = "'src=\"(.*?)\"'si";
+    preg_match($string_pattern_poster_src, $regextextposter[0], $postersrc);
+
+    //extracts movie plot text
+    $string_pattern_summary_text = "'<div class=\"summary_text\" itemprop=\"description\">(.*?)</div>'si";
+    preg_match($string_pattern_summary_text, $homepage, $summary_text);
+    @$movie_infos = array(
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: remove @

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+/**
+ * Returns movie title, link to the posterlink, the source of the poster image and the movie plot.
+ *
+ * @params string $url
+ *   The movie url of an imdb movie page
+ */
+function get_imdb_movie_infos($url) {
+    @$homepage = file_get_contents($url);
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: remove @

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+    return $page;
+}
+
+/**
+ * Implements hook_theme(). 
+ */
+function imdb_movie_info_crawler_theme() {
+    return array(
+      'show_movie_infos' => array(
+        'variables' => array('infos' => []),
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module:95]: add defaults

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php [hunk 17]
```diff
@@ -0,0 +1,10 @@
+<h1><?php print $infos['title']; ?></h1>
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php]: add check

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php [hunk 18]
```diff
@@ -0,0 +1,10 @@
...
+<div>
+    <a href="<?php print $infos['posterlink']; ?>" >
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php]: for links use l function

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php [hunk 19]
```diff
@@ -0,0 +1,10 @@
...
+           <img alt='movie poster'  src="<?php print $infos['postersrc']; ?>"/>
+    </a>
+</div>
+</br>
+<p><?php print $infos['plot']; ?></p>
+</br>
+<h1><a style="color: red; text-decoration: underline; font-size: 200%;" href='./Imdb_Crawler'>again</a></h1>
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php]: remove inline styles

File: nbproject/private/private.xml [hunk 20]
```diff
@@ -1,4 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (carsten-propeople) [inline nbproject/private/private.xml]: add this file to .gitignore

**Review by carsten-propeople — CHANGES_REQUESTED** [review]

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module [hunk 21]
```diff
@@ -45,38 +45,41 @@ function imdb_movie_info_crawler_form($form, &$form_state) {
...
  */
 function imdb_movie_info_crawler_form_validate($form, &$form_state) {
     $submitted_url = $form_state['values']['movie_url'];
-    $string_pattern_url = "'http://www.imdb.com/title/tt'si";
+    $string_pattern_url = "'^http://www.imdb.com/title/tt's";
     $valid_url = preg_match($string_pattern_url, $submitted_url);
     if (!$valid_url) {
-        drupal_set_message("Please enter a valid url!");
-        $form_state['rebuild'] = TRUE;
+        form_set_error("movie_url", "Please enter a valid url!");
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module]: t() is missing

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php [hunk 22]
```diff
@@ -1,10 +1,12 @@
-<h1><?php print $infos['title']; ?></h1>
-<div>
-    <a href="<?php print $infos['posterlink']; ?>" >
-           <img alt='movie poster'  src="<?php print $infos['postersrc']; ?>"/>
-    </a>
-</div>
+<h1><?php print !empty($infos['title']) ? $infos['title'] : "No title found!" ; ?></h1>
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php]: t() is missing

File: drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php [hunk 23]
```diff
@@ -1,10 +1,12 @@
...
+        </a>
+    </div>
+<?php endif; ?>    
 </br>
-<p><?php print $infos['plot']; ?></p>
+<p><?php print !empty($infos['plot']) ? $infos['plot'] : "" ; ?></p>
 </br>
-<h1><a style="color: red; text-decoration: underline; font-size: 200%;" href='./Imdb_Crawler'>again</a></h1>
\ No newline at end of file
+<h1><?php print l("AGAIN", "/imdb_crawler", array('attributes' => array('class' => 'back-link'))); ?></h1>
```
Comment (carsten-propeople) [inline drupal/web/sites/all/modules/costum/imdb movie info crawler/templates/movie_infos.tpl.php]: t() is missing

File: nbproject/private/private.xml [hunk 24]
```diff
@@ -3,13 +3,12 @@
     <editor-bookmarks xmlns="http://www.netbeans.org/ns/editor-bookmarks/2" lastBookmarkId="0"/>
```
Comment (carsten-propeople) [inline nbproject/private/private.xml]: this file needs to be removed from the index

**Review by carsten-propeople — CHANGES_REQUESTED** [review]

**Comment by Lucas5Lucas** [comment]
@carsten-propeople changes have been made

**Pull request merged**